hazenlib/tasks/task_acr_ghosting.py

```diff
@@ -0,0 +1,194 @@
+"""
+ACR Ghosting
+
+Calculates uniformity for slice 7 of the ACR phantom.
+
+This script calculates the percentage signal ghosting in accordance with the ACR Guidance.
+This is done by first defining a large 200cm2 ROI before placing 10cm2 elliptical ROIs outside the phantom along the
+cardinal directions. The results are also visualised.
+
+Created by Yassine Azma
+yassine.azma@rmh.nhs.uk
+
+14/11/2022
+"""
+
+import sys
+import traceback
+import os
+import numpy as np
+import skimage.morphology
+
+from hazenlib.HazenTask import HazenTask
+
+
+class ACRGhosting(HazenTask):
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def run(self) -> dict:
+        results = {}
+        z = []
+        for dcm in self.data:
+            z.append(dcm.ImagePositionPatient[2])
+
+        idx_sort = np.argsort(z)
+
+        for dcm in self.data:
+            if dcm.ImagePositionPatient[2] == z[idx_sort[6]]:
+                try:
+                    result = self.get_signal_ghosting(dcm)
+                except Exception as e:
+                    print(f"Could not calculate the percent-signal ghosting for {self.key(dcm)} because of : {e}")
+                    traceback.print_exc(file=sys.stdout)
+                    continue
+
+                results[self.key(dcm)] = result
+
+        results['reports'] = {'images': self.report_files}
+
+        return results
+
+    def centroid_com(self, dcm):
+        # Calculate centroid of object using a centre-of-mass calculation
+        thresh_img = dcm > 0.25 * np.max(dcm)
+        open_img = skimage.morphology.area_opening(thresh_img, area_threshold=500)
+        bhull = skimage.morphology.convex_hull_image(open_img)
+        coords = np.nonzero(bhull)  # row major - first array is columns
+
+        sum_x = np.sum(coords[1])
+        sum_y = np.sum(coords[0])
+        cxy = sum_x / coords[0].shape, sum_y / coords[1].shape
+
+        cxy = [cxy[0].astype(int), cxy[1].astype(int)]
+        return bhull, cxy
+
+    def get_signal_ghosting(self, dcm):
+        img = dcm.pixel_array  # extract image slice of interest
```

Comment:
What does the comment refer to? Is this not just taking the image values?

Reply:
That's an old comment from my MATLAB script where I would work directly from a 3D array. I'll remove that - what that line does is apparent anyway.